mergify_engine/context.py

```diff
@@ -63,25 +63,74 @@ class Installation:
     subscription: subscription.Subscription
     client: github.GithubInstallationClient
 
+    repositories: "typing.Dict[github_types.GitHubRepositoryName, Repository]" = (
+        dataclasses.field(default_factory=dict)
+    )
+
     @property
     def stream_name(self) -> "worker.StreamNameType":
         return typing.cast(
             "worker.StreamNameType", f"stream~{self.owner_login}~{self.owner_id}"
         )
 
+    def get_repository(self, name: github_types.GitHubRepositoryName) -> "Repository":
+        if name not in self.repositories:
+            repository = Repository(self, name)
+            self.repositories[name] = repository
+        return self.repositories[name]
+
+
+@dataclasses.dataclass
+class Repository(object):
+    installation: Installation
+    name: str
+    pull_contexts: "typing.Dict[github_types.GitHubPullRequestNumber, Context]" = (
+        dataclasses.field(default_factory=dict)
+    )
+
+    @property
+    def base_url(self) -> str:
+        """The URL prefix to make GitHub request."""
+        return f"/repos/{self.installation.owner_login}/{self.name}"
```

Comment:
This is actually `github_types.GitHubRepository['url']` so I'm wondering if we really need this.
If it's possible to use the GitHubRepository URL that makes it easier to support different URL scheme etc :)

Reply:
This url is currently not forwarded from the webhook to the worker.